Source/Core/DiscIO/NANDImporter.cpp

```diff
@@ -131,27 +130,27 @@ std::string NANDImporter::GetPath(const NANDFSTEntry& entry, const std::string&
 
 std::string NANDImporter::FormatDebugString(const NANDFSTEntry& entry)
 {
-  return fmt::format(
-      "{:12.12} {:#04x} {:#04x} {:#06x} {:#06x} {:#010x} {:#06x} {:#06x} {:#06x} {:#010x}",
-      entry.name, entry.mode, entry.attr, entry.sub, entry.sib, entry.size, entry.x1, entry.uid,
-      entry.gid, entry.x3);
+  return fmt::format("{:12.12} {:#010b} {:#04x} {:#06x} {:#06x} {:#010x} {:#010x} {:#06x} {:#010x}",
+                     entry.name, entry.mode, entry.attr, entry.sub, entry.sib, entry.size,
+                     entry.uid, entry.gid, entry.x3);
 }
 
 void NANDImporter::ProcessEntry(u16 entry_number, const std::string& parent_path)
 {
   NANDFSTEntry entry;
-  memcpy(&entry, &m_nand[m_nand_fst_offset + sizeof(NANDFSTEntry) * Common::swap16(entry_number)],
+  memcpy(&entry, &m_nand[m_nand_fst_offset + sizeof(NANDFSTEntry) * entry_number],
          sizeof(NANDFSTEntry));
 
   if (entry.sib != 0xffff)
     ProcessEntry(entry.sib, parent_path);
 
-  if ((entry.mode & 3) == 1)
+  Type type = static_cast<Type>(entry.mode & 3);
+  if (type == Type::File)
     ProcessFile(entry, parent_path);
-  else if ((entry.mode & 3) == 2)
+  else if (type == Type::Directory)
     ProcessDirectory(entry, parent_path);
   else
-    ERROR_LOG_FMT(DISCIO, "Unknown mode: {}", FormatDebugString(entry));
+    ERROR_LOG_FMT(DISCIO, "Ignoring unknown entry type {}", static_cast<int>(type));
```

Comment:
This removes some of the information from the log message, which might make it harder to debug.  I'd recommend `"Ignoring unknown entry type for {}"` and provide `entry` as the argument.